hoomd/hpmc/shape_move.py

```diff
@@ -0,0 +1,273 @@
+        mc.shape["A"] = dict(vertices=verts)
+        elastic_move = hoomd.hpmc.shape_move.Elastic(stiffness=10, mc)
+        elastic_move.stiffness = 100
+        elastic_move.reference_shape["A"] = verts
+
+    Attributes:
+        stiffness (:py:mod:`hoomd.variant.Variant`): Shape stiffness against
+            deformations.
+
+        reference_shape (`TypeParameter` [``particle type``, `dict`]): Reference
+            shape against to which compute the deformation energy.
+
+        move_probability (`float`): Fraction of scale to shear moves.
+    """
+
+    _suported_shapes = {'ConvexPolyhedron', 'Ellipsoid'}
+
+    def __init__(self, stiffness, mc, move_probability=0.5):
+
+        super().__init__(move_probability)
+        param_dict = ParameterDict(stiffness=hoomd.variant.Variant)
+        param_dict["stiffness"] = stiffness
+        self._param_dict.update(param_dict)
+        self._add_typeparam(self._get_shape_param(mc))
+
+    def _get_shape_param(self, mc):
+        if isinstance(mc, hoomd.hpmc.integrate.HPMCIntegrator):
+            cls = mc.__class__
+        else:
+            cls = mc
+        if cls.__name__ not in self._suported_shapes:
+            raise ValueError(f"Unsupported integrator type {cls}. Supported "
+                             f"types are {self._suported_shapes}")
+        # Class can only be used for this type of integrator now.
+        self._suported_shapes = {cls.__name__}
+        shape = cls().shape
+        shape.name = "reference_shape"
+        return shape
+
+    def _attach(self):
+        integrator = self._simulation.operations.integrator
+        if isinstance(integrator, integrate.Ellipsoid):
+            for shape in integrator.shape.values():
+                if not numpy.allclose(
+                    (shape["a"], shape["b"], shape["c"]), shape["a"]):
+                    raise ValueError("This updater only works when a=b=c.")
+        super()._attach()
+
+
+class ShapeSpace(ShapeMove):
+    """Apply shape moves in a :math:`N` dimensional shape space.
+
+    Args:
+        callback (``callable`` [`str`, `list`], `dict` ]): The python function
+            that will be called to map the given shape parameters to a shape
+            definition. The function takes the particle type and a list of
+            parameters as arguments and return a dictionary with the shape
+            definition whose keys **must** match the shape definition of the
+            integrator:
+
+                ``callable[[str, list], dict]``
+
+            Note that there is no type validation of the callback.
+
+        move_probability (`float`, optional): Average fraction of shape
+            parameters to change each timestep (**default**: 1).
+
+    .. rubric:: Shape support.
+
+    The following shapes are supported:
+        * `hoomd.hpmc.integrate.ConvexPolyhedron`
+        * `hoomd.hpmc.integrate.ConvexSpheropolyhedron`
+        * `hoomd.hpmc.integrate.Ellipsoid`
```

Comment:
```suggestion
    The following shapes are supported:
    
    * `hoomd.hpmc.integrate.ConvexPolyhedron`
    * `hoomd.hpmc.integrate.ConvexSpheropolyhedron`
    * `hoomd.hpmc.integrate.Ellipsoid`
```
Not a definition list.